pkg/api/v1alpha1/cluster_types.go

```diff
@@ -167,6 +182,8 @@ type WorkerNodeGroupConfiguration struct {
 	Count int `json:"count,omitempty"`
 	// MachineGroupRef defines the machine group configuration for the worker nodes.
 	MachineGroupRef *Ref `json:"machineGroupRef,omitempty"`
+	// Taints define the set of taints to be applied on worker nodes
+	Taints []corev1.Taint `json:"taints,omitempty"`
 }
 
 func generateWorkerNodeGroupKey(c WorkerNodeGroupConfiguration) (key string) {
```

Comment:
need to update the `WorkerNodeGroupConfigurationsSliceEqual` method as well

Reply:
Will do it when I work on the node group changes.